src/libsyntax_pos/lib.rs

```diff
@@ -112,12 +112,13 @@ impl std::fmt::Display for FileName {
         match *self {
             Real(ref path) => write!(fmt, "{}", path.display()),
             Macros(ref name) => write!(fmt, "<{} macros>", name),
-            QuoteExpansion => write!(fmt, "<quote expansion>"),
-            MacroExpansion => write!(fmt, "<macro expansion>"),
-            Anon => write!(fmt, "<anon>"),
-            ProcMacroSourceCode => write!(fmt, "<proc-macro source code>"),
-            CfgSpec => write!(fmt, "cfgspec"),
-            CliCrateAttr => write!(fmt, "<crate attribute>"),
+            QuoteExpansion(fingerprint) => write!(fmt, "<{} quote expansion>", fingerprint),
+            MacroExpansion(fingerprint) => write!(fmt, "<{} macro expansion>", fingerprint),
+            Anon(fingerprint) => write!(fmt, "<{} anon>", fingerprint),
+            ProcMacroSourceCode(fingerprint) =>
+                write!(fmt, "<{} proc-macro source code>", fingerprint),
+            CfgSpec(fingerprint) => write!(fmt, "<{} cfgspec>", fingerprint),
+            CliCrateAttr(fingerprint) => write!(fmt, "<{} crate attribute>", fingerprint),
```

Comment:
Since these are strings visible to the user, I would not add the fingerprint to them. And we don't need them for correctness here.

Reply:
Noted. I just fixed this and rebased with master to resolve conflicts. There will still be a test failure(s) -- something related to converting expression to string and then back to expression and making sure they are the same. Will look into it but my local git setup is now pretty screwed and nothing will build because one of the submodules is missing things